src/ucs/async/async_int.h

```diff
@@ -20,18 +21,30 @@ struct ucs_async_handler {
     ucs_async_event_cb_t       cb;      /* Callback function */
     void                       *arg;    /* Callback argument */
     ucs_async_context_t        *async;  /* Async context for the handler. Can be NULL */
-    volatile uint32_t          missed; /* Protect against adding to miss queue multiple times */
-    ucs_async_handler_t        *next;
+    volatile uint32_t          missed;  /* Protect against adding to miss queue multiple times */
+    volatile uint32_t          refcount;
 };
 
 
 /**
  * Dispatch event coming from async context.
  *
- * @param id         Handler to dispatch.
+ * @param id         Array of event IDs to dispatch.
+ * @param count      Number of events
  * @param from_async Whether the handler is called from async context or main context.
```

Comment:
can we just some sort of context id to the function and the function will check if this is async or main thread ?